sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java

```diff
@@ -29,24 +29,98 @@
   private static final long INVALID_ID = 0;
 
   @Override
-  public SpanId generateSpanId() {
+  public CharSequence generateSpanId() {
     long id;
     ThreadLocalRandom random = ThreadLocalRandom.current();
     do {
       id = random.nextLong();
     } while (id == INVALID_ID);
-    return new SpanId(id);
+    return new SpanIdWrapper(id);
   }
 
   @Override
-  public TraceId generateTraceId() {
+  public CharSequence generateTraceId() {
     long idHi;
     long idLo;
     ThreadLocalRandom random = ThreadLocalRandom.current();
     do {
       idHi = random.nextLong();
       idLo = random.nextLong();
     } while (idHi == INVALID_ID && idLo == INVALID_ID);
-    return new TraceId(idHi, idLo);
+    return new TraceIdWrapper(idHi, idLo);
+  }
+
+  private static final class TraceIdWrapper implements CharSequence {
```

Comment:
Ah now I see why char sequence :) Interesting optimization but considering how common it is to use the trace id as a string, I think I wouldn't do this. It does make the current microbenchmarks difficult to interpret but such is life. It's sort of obvious that wrapping a long is going to be faster if only generating random longs, so the only reason we're even trying strings is since we hope for it to be mostly a wash given the other usages of the trace id.

I think we can have two benchmarks that should be comparable for this PR. Incidentally they sound more like a pipeline than just generating the span :)

1) random generate trace id, span id, fill trace header, fill into otlp proto
2) extract trace id from header, random generate span id, fill trace header, fill into otlp proto

These seem like they'd give us the info to make a decision on a trade-off. I'm hoping the numbers are very similar for both, especially after buffer recycling

Reply:
yeah, I did this mostly as an experiment to see how it felt to have a custom `CharSequence` implementation in the mix. It definitely optimizes for the internally-sourced trace codepath, rather than the externally-sourced trace codepath. It's also only used in the internally-sourced trace codepath because you wouldn't need to generate one when you're grabbing it from the external request.

Yes, more complete benchmarks could definitely give a better overall picture. I have historical allocation benchmarks for the one SpanPipelineBenchmark class, but hadn't been capturing other benchmarks over time. I'll spend some time gathering some more data, as well as re-using the char[] to further reduce allocations.

Thanks for taking a look! Your feedback is very helpful.